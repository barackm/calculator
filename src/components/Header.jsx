import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <nav className="navbar navbar-expand-lg navbar-light fixed-top py-3" id="mainNav">
    <div className="container px-4 px-lg-5">
      <Link className="navbar-brand" to="/">Mathmaticians</Link>
      <button className="navbar-toggler navbar-toggler-right" type="button" data-bs-toggle="collapse" data-bs-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation"><span className="navbar-toggler-icon" /></button>
      <div className="collapse navbar-collapse" id="navbarResponsive">
        <ul className="navbar-nav ms-auto my-2 my-lg-0">
          <li className="nav-item"><Link className="nav-link" to="/">About</Link></li>
          <li className="nav-item"><Link className="nav-link" to="/">Services</Link></li>
          <li className="nav-item"><Link className="nav-link" to="/">Portfolio</Link></li>
          <li className="nav-item"><Link className="nav-link" to="/">Contact</Link></li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Header;
