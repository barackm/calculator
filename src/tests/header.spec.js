import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';

describe('Header Component', () => {
  it('Should match spanshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
