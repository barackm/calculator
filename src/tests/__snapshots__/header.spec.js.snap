// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should match spanshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light fixed-top py-3"
  id="mainNav"
>
  <div
    className="container px-4 px-lg-5"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Mathmaticians
    </a>
    <button
      aria-controls="navbarResponsive"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-bs-target="#navbarResponsive"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarResponsive"
    >
      <ul
        className="navbar-nav ms-auto my-2 my-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/quotes"
            onClick={[Function]}
          >
            Quotes
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
